lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -3712,8 +3713,16 @@ void SwiftASTContext::LoadModule(swift::ModuleDecl *swift_module,
         all_dlopen_errors.GetData());
   };
 
-  for (auto import : swift::namelookup::getAllImports(swift_module)) {
+  auto imported_modules = swift::namelookup::getAllImports(swift_module);
+
+  Progress progress(llvm::formatv("Loading module {0} dependencies",
+                                  swift_module->getNameStr().data()),
+                    imported_modules.size());
+
+  size_t completion = 0;
+  for (auto import : imported_modules) {
     import.importedModule->collectLinkLibraries(addLinkLibrary);
+    progress.Increment(++completion);
```

Comment:
I think this one is too deep inside. What do think about instead moving this progress meter to
SwiftASTContext::CacheUserImports() and SwiftASTContext::GetCompileUnitImportsImpl() which are calling into this function?

Reply:
By "to deep" I mean that this loop happens *after* the module has already been loaded and is comparatively quick.
What happens if a Progress created while another Progress is already in flight? If that's handled gracefully, we can keep this too and just add the outer loop.